pkg/kubectl/resource/visitor.go

```diff
@@ -691,7 +692,23 @@ func (v FilteredVisitor) Visit(fn VisitorFunc) error {
 	})
 }
 
-func FilterBySelector(s labels.Selector) FilterFunc {
+func FilterByLabelSelector(s labels.Selector) FilterFunc {
+	return func(info *Info, err error) (bool, error) {
+		if err != nil {
+			return false, err
+		}
+		a, err := meta.Accessor(info.Object)
+		if err != nil {
+			return false, err
+		}
+		if !s.Matches(labels.Set(a.GetLabels())) {
+			return false, nil
+		}
+		return true, nil
+	}
+}
+
+func FilterByFieldSelector(s fields.Selector) FilterFunc {
```

Comment:
Not sure if this is correct. Why does the field selector use labels here?

Reply:
@kargakis Thanks for catching this. I remove this function in a new commit. PTAL.

Currently we don't need such function.